src/components/NutrientForm.js

```diff
@@ -0,0 +1,156 @@
+    setCarbProportion(calcProportion(carbKcal, totalKcal))
+  }
+
+  const calcDryMatterBasis = ({ protein, fat, moisture, fiber, ash, totalCalories }) => {
+    let carb = calcCarb(protein, fat, moisture, fiber, ash)
+    // check if carb is negative
+    if (carb < 0) {
+      const carbKcal = totalCalories - protein * 3.5 - fat * 8.5
+      carb = carbKcal / 3.5
+    }
+
+    const dryMatter = 100 - moisture
+    setDryProtein(calcProportion(protein, dryMatter))
+    setDryFat(calcProportion(fat, dryMatter))
+    setDryCarb(calcProportion(carb, dryMatter))
+  }
+
+  const calcCalciumToPhosphorusRatio = ({ calcium, phosphorus }) => {
+    setCalciumToPhosphorusRatio(calcium / phosphorus)
+  }
+
+  useEffect(() => {
+    calcMetabolizableEnergy(inputs)
+    calcDryMatterBasis(inputs)
+    calcCalciumToPhosphorusRatio(inputs)
+  }, [inputs])
+
+  return (
+    <Box
+      component="form"
+      sx={{
+        '& .MuiTextField-root': { m: 1, width: 200 },
+      }}
+      noValidate
+      autoComplete="off"
+    >
+      <FormControl fullWidth>
+        <InputField
+          label={INPUTS_INFO.TOTAL_CALORIES.label}
+          type="Number"
+          name={INPUTS_INFO.TOTAL_CALORIES.name}
+          value={inputs[INPUTS_INFO.TOTAL_CALORIES]}
+          onChange={handleInputsChange}
+        />
+        <InputField
+          label={INPUTS_INFO.PROTEIN.label}
+          type="Number"
+          name={INPUTS_INFO.PROTEIN.name}
+          value={inputs[INPUTS_INFO.PROTEIN]}
+          onChange={handleInputsChange}
+        />
+        <InputField
+          label={INPUTS_INFO.FAT.label}
+          type="Number"
+          name={INPUTS_INFO.FAT.name}
+          value={inputs[INPUTS_INFO.FAT]}
+          onChange={handleInputsChange}
+        />
+        <InputField
+          label={INPUTS_INFO.MOISTURE.label}
+          type="Number"
+          name={INPUTS_INFO.MOISTURE.name}
+          value={inputs[INPUTS_INFO.MOISTURE]}
+          onChange={handleInputsChange}
+        />
+        <InputField
+          label={INPUTS_INFO.FIBER.label}
+          type="Number"
+          name={INPUTS_INFO.FIBER.name}
+          value={inputs[INPUTS_INFO.FIBER]}
+          onChange={handleInputsChange}
+        />
+        <InputField
+          label={INPUTS_INFO.ASH.label}
+          type="Number"
+          name={INPUTS_INFO.ASH.name}
+          value={inputs[INPUTS_INFO.ASH]}
+          onChange={handleInputsChange}
+        />
+        <InputField
+          label={INPUTS_INFO.CALCIUM.label}
+          type="Number"
+          name={INPUTS_INFO.CALCIUM.name}
+          value={inputs[INPUTS_INFO.CALCIUM]}
+          onChange={handleInputsChange}
+        />
+        <InputField
+          label={INPUTS_INFO.PHOSPHORUS.label}
+          type="Number"
+          name={INPUTS_INFO.PHOSPHORUS.name}
+          value={inputs[INPUTS_INFO.PHOSPHORUS]}
+          onChange={handleInputsChange}
+        />
+      </FormControl>
```

Comment:
I think it will be much easier if you can add a button and calculate everything after submitting the form. There's no point in tracking every input field's status before filling up the form.